Orange/widgets/visualize/owpythagorastree.py

```diff
@@ -0,0 +1,387 @@
+
+    Square component to draw as components for the Pythagoras tree. It is
+    initialized with a given center QPointF, the main side length, and the
+    initial angle the sqaure is to be rendered with.
+
+    Parameters
+    ----------
+    center : Point
+        The central point of the square
+    length : float
+        The length of the square sides
+    angle : float
+        The initial angle at which the square will be rotated (in rads)
+
+    Attributes
+    ----------
+    center
+    length
+    angle
+
+    """
+
+    def __init__(self, center, length, angle, parent=None):
+        self._center_point = center
+        self.center = QtCore.QPointF(*center)
+        self.length = length
+        self.angle = angle
+        super().__init__(self._get_rect_attributes(), parent)
+        # no need to perform extra transformations where they won't be seen
+        if angle % 90 != 0:
+            self.setTransformOriginPoint(self.boundingRect().center())
+            self.setRotation(math.degrees(angle))
+
+    def _get_rect_attributes(self):
+        """
+        Compute the QRectF that a QGraphicsRect needs to be rendered with the
+        data passed down in the constructor.
+        """
+        height = width = self.length
+        x = self.center.x() - self.length / 2
+        y = self.center.y() - self.length / 2
+        return QtCore.QRectF(x, y, height, width)
+
+
+class TreeNode:
+    """A node in the tree structure used to represent the tree adapter
+
+    Parameters
+    ----------
+    square : Square
+        The square the represents the tree node.
+    parent : TreeNode
+        The parent of the current node.
+    children : list of TreeNode, optional
+        All the children that belong to this node.
+
+    """
+    def __init__(self, square, parent, children=[]):
+        self.square = square
+        self.parent = parent
+        self.children = children
+
+
+class PythagorasTree:
+    def pythagoras_tree(self, tree, node=0,
+                        square=Square(Point(0, 0), 200, 0)):
+        """Get the Pythagoras tree representation in a graph like view.
+
+        Constructs a graph using TreeNode into a tree structure. Each node in
+        graph contains the information required to plot the the tree.
+
+        Parameters
+        ----------
+        tree : TreeAdapter
+            A tree adapter instance where the original tree is stored.
+        node : int, optional
+            The node label, uses 0 for the root node (the default is 0).
+        square : Square, optional
+            The initial square where the drawing will start from.
+
+        Returns
+        -------
+        TreeNode
+            The root node which contains the rest of the tree.
+
+        """
+        children = []
+        if tree.has_children(node):
+            left_child = tree.left_child(node)
+            right_child = tree.right_child(node)
+
+            w1, w2 = tree.weight(left_child), tree.weight(right_child)
```

Comment:
I tried to follow the `tree.weight` but I can't see if it's possible or for both `w1` and `w2` to be zero. So if that's the case, can you make sure this does not fail with `ZeroDivisionError`. And if it's not possible, I'd be happy if you can give a quick reason. Thanks.